backends/ebpf/ebpfenv.py

```diff
@@ -0,0 +1,162 @@
+        self.verbose = verbose       # do we want to be chatty?
+
+    def ns_init(self):
+        """ Initialize the namespace. """
+        cmd = "ip netns add %s" % self.ns_name
+        errmsg = "Failed to create namespace %s :" % self.ns_name
+        result = run_timeout(True, cmd, TIMEOUT,
+                             self.outputs, errmsg)
+        self.ns_exec("ip link set dev lo up")
+        return result
+
+    def ns_del(self):
+        """ Delete the namespace. """
+        cmd = "ip netns del %s" % self.ns_name
+        errmsg = "Failed to delete namespace %s :" % self.ns_name
+        return run_timeout(True, cmd, TIMEOUT,
+                           self.outputs, errmsg)
+
+    def get_ns_prefix(self):
+        """ Return the command prefix for the namespace of this bridge class.
+            """
+        return "ip netns exec %s" % self.ns_name
+
+    def ns_exec(self, cmd_string):
+        """ Run and execute an isolated command in the namespace. """
+        prefix = self.get_ns_prefix()
+        # bash-c allows us to run multiple commands at once
+        cmd = "%s bash -c \"%s\"" % (prefix, cmd_string)
+        errmsg = "Failed to run command in namespace %s:" % self.ns_name
+        return run_timeout(self.verbose, cmd, TIMEOUT,
+                           self.outputs, errmsg)
+
+    def ns_proc_open(self):
+        """ Open a bash process in the namespace and return the handle """
+        cmd = self.get_ns_prefix() + " /bin/bash "
+        return open_process(self.verbose, cmd, self.outputs)
+
+    def ns_proc_write(self, proc, cmd):
+        """ Allows writing of a command to a given process. The command is NOT
+            yet executed. """
+        report_output(self.outputs["stdout"],
+                      self.verbose, "Writing %s " % cmd)
+        try:
+            proc.stdin.write(cmd)
+        except IOError as e:
+            err_text = "Error while writing to process"
+            report_err(self.outputs["stdout"],
+                       self.verbose, err_text, e)
+            return FAILURE
+        return SUCCESS
+
+    def ns_proc_append(self, proc, cmd):
+        """ Append a command to an open process. """
+        return self.ns_proc_write(proc, " && " + cmd)
+
+    def ns_proc_close(self, proc):
+        """ Close and actually run the process in the namespace. Returns the
+            exit code. """
+        errmsg = ("Failed to execute the command"
+                  " sequence in namespace %s" % self.ns_name)
+        return run_process(self.verbose, proc, TIMEOUT, self.outputs, errmsg)
+
+    def _configure_bridge(self, br_name):
+        """ Set the bridge active and load the qdisc. We also disable IPv6 to
+            avoid ICMPv6 spam. """
+        cmd = "ip link set dev %s up && " % br_name
+        # Prevent the broadcasting of ipv6 link discovery messages
+        cmd += "sysctl -w net.ipv6.conf.all.disable_ipv6=1 && "
+        cmd += "sysctl -w net.ipv6.conf.default.disable_ipv6=1 && "
```

Comment:
I tried both setting accept_ra and fowarding to zero, neither worked so I had to disable ipv6 entirely. I hope this does not conflict with the sending of actual ipv6 packets. It should not since we write directly to the interface.

Reply:
I've seen this problem before. This looks fine for now.